early-attack-detector-py/src/agent.py

```diff
@@ -222,12 +243,12 @@ def detect_malicious_contract_tx(
                         finding.metadata["funding_labels"] = ','.join(funding_labels)
                     if float(finding.metadata['model_score']) >= MODEL_PRECISION_THRESHOLD:
                         finding.metadata['high_precision_model'] = True
-                    # If the model is working in high precision, or it has a 1-day funding alert, we raise the alert and continue
-                    if 'high_precision_model' in finding.metadata.keys() or 'funding_labels' in finding.metadata.keys():
+                    # If the model is working in high precision, has a 1-day funding alert, or the txn `from` is in the TP list, we raise the alert and continue
+                    if 'high_precision_model' in finding.metadata.keys() or 'funding_labels' in finding.metadata.keys() or 'known_past_attacker' in finding.metadata.keys():
                         all_findings.append(finding)
                         continue
                 # This should only trigger in beta and if no critical alert has been raised
-                if BETA:
+                elif BETA:
```

Comment:
Testing the bot I believe this change doesn't cover the case of the score exceeding the `MODEL_THRESHOLD` (i.e. Critical severity Finding) but then not raising a Critical alert in the _`if finding.severity == FindingSeverity.Critical`_ branch (due to the other checks) [The use of `continue` in the if branch prevents the branch `if BETA` from being entered when a Critical alert has been raised anyway]

Reply:
At first, I wasn't sure I understood.

You mean if `if finding.severity == FindingSeverity.Critical:` passes, because if `if len(funding_labels) > 0:` and `if float(finding.metadata['model_score']) >= MODEL_PRECISION_THRESHOLD:` don't pass, there may not be a `Critical` alert created? Because this `if 'high_precision_model' in finding.metadata.keys() or 'funding_labels' in finding.metadata.keys() or 'known_past_attacker' in finding.metadata.keys():` check wouldn't pass.

But _if_ `if 'high_precision_model' in finding.metadata.keys() or 'funding_labels' in finding.metadata.keys() or 'known_past_attacker' in finding.metadata.keys():` _did_ pass, the `if BETA` wouldn't be entered because after `all_findings.append(finding)`, it'll just `continue`, and no enter the `if BETA`?

I see now. Reverting the change. 👍